clustermesh/clustermesh.go

```diff
@@ -826,6 +829,18 @@ func (k *K8sClusterMesh) extractAccessInformation(ctx context.Context, client k8
 		}
 	}
 
+	tunnelProtocol := ""
+	if cm.Data[configNameRoutingMode] == "tunnel" {
```

Comment:
nit: Should we check the software version vs relying solely on the config option. ie could surface that the flag is going to be deprecated back to the our users...

Reply:
I don't it makes much sense to make reading of the flag version dependent (when writing the flag, then absolutely, we should check the version and that's what the PR already does). At least at the moment, all released versions of Cilium understand the old version of the flag. Once the flag has been removed (which may or may not happen in v1.15), that might change, but we can only remove the flag _after_ we've merged this PR and fixed our CI.

Surfacing the flag is planned (see the comment I've linked in the OP) - but it should be flagged during installation time. And maybe when running `cilium status`. But raising this while the user is trying to enable clustermesh (a separate task that has nothing to do with tunneling) I think is not the right thing to do, because at that point it's already "too late" and would be confusing the the user, because that's not what they are focusing on right now.